lib/ansible/modules/hostname.py

```
@@ -297,6 +297,24 @@ def __init__(self, module):
         super(SystemdStrategy, self).__init__(module)
         self.hostnamectl_cmd = self.module.get_bin_path(self.COMMAND, True)
 
+    def _scrub_hostname(self, name):
+        """
+        Clean hostname by replacing invalid characters with dashes
```

Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example. Also, add tests for this change.

---

I would not just change the user input w/o notice, a better change is giving them more information on why the hostname is not valid

---

Hi,
Thanks for first review.

I just added fragment, and i will look at the testing implementation.
Regarding of user input, i just used existing code of DarwinStrategy which not warn about change, maybe we can use warning and continue the execution ?

---

Sigh, I really dislike this but since it already exists ... lets see if the following makes sense:
- make this a generic option:  sanitize? boolean, defaults to true, uses a class specific 'safe' char and 'unsafe' list of chars. When not set (default value) it adds warning.

---

Thanks for your answer.
Regarding DarwinStrategy, would you like me to apply the same behavior there (generic sanitize option and warning)?

Also, concerning RedHatStrategy, the main limitation is that we cannot distinguish between static, transient, and pretty hostnames in the playbook when using it, so if i want set test_erwanclx, it is valid for pretty, not for transient.

---

Then I would go back to 'invalid names are errors', for example, using a name longer than 64 chars on systemd. I would even change the darwin strategy to conform to this

---

Thanks for your feedback.
If I understand correctly, you're suggesting that we should treat invalid names as errors across all strategies (including DarwinStrategy) and not apply any sanitization or warnings ?

---

I understand your point of view, but how can we deal with special character in hostname with SystemdStrategy ?
Actually we can't specify the hostname type (static, pretty, transient), so for example, if i want pass `test_erwanclx` i will struggle on error, however it is valid for pretty.

---

attempt the more stringent first, let that fail and avoid other updates, otherwise it would require a rollback

---

So, i close the PR with no merge or we stay with a bypass like i did ?